.github/workflows/website-deploy.yml

```diff
@@ -16,8 +16,47 @@ concurrency:
   cancel-in-progress: true
 
 jobs:
+  build-corelib-docs:
+    runs-on: ubuntu-latest
+    env:
+      MDBOOK_VERSION: 0.4.31
+    steps:
+      - uses: actions/checkout@v4
+      - uses: dtolnay/rust-toolchain@stable
+      - uses: software-mansion/setup-scarb@v1
+      - name: Install mdBook
+        run: cargo install --version ${MDBOOK_VERSION} mdbook
+      - name: Init Scarb to cache corelib
+        id: scarb-init
+        run: |
+          mkdir -p $HOME/pkg
+          pushd $HOME/pkg
+          scarb init --no-vcs
+          echo "CAIRO_VERSION_PATH=$(scarb metadata --format-version 1 | jq -r '.app_version_info.cairo | if .commit_info != null then .commit_info.short_commit_hash else "v\(.version)" end')" >> $GITHUB_OUTPUT
+        env:
+          SCARB_INIT_TEST_RUNNER: cairo-test
+      - name: Extract corelib from Scarb cache
+        run: |
+          mkdir -p core
+          cp -r $(scarb cache path)/registry/std/${{ steps.scarb-init.outputs.CAIRO_VERSION_PATH }}/core/* ./core
```

Comment:
This can be read from metadata directly though, no need to construct the path here.